### PR TITLE
vnc: change password file to use guid hash

### PR DESCRIPTION
**Pull request by jsorg71**
This is cleanup of vncpassfile branch

### CODE REVIEW

File: vnc/vnc.c [hunk 1]
```diff
@@ -1083,7 +1094,24 @@ lib_mod_connect(struct vnc *v)
...
+                        char guid_str[64];
+                        char *pguid_str;
+                        int index;
+                        pguid_str = guid_str;
+                        for (index = 0; index < 16; index++)
+                        {
+                            g_snprintf(pguid_str, 4, "%2.2x", v->guid[index]);
+                            pguid_str += 2;
+                        }
+                        guid_str[32] = 0;
```
Comment (proski) [inline vnc/vnc.c]: This code is used twice, please refactor

File: sesman/session.c [hunk 2]
```diff
@@ -688,18 +686,31 @@ session_start_fork(int width, int height, int bpp, char *username,
...
+                    g_sprintf(geometry, "%d", s->height);
                     g_setenv("XRDP_START_HEIGHT", geometry, 1);
 
                     /* fire up Xorg */
                     g_execvp(xserver, pp1);
                 }
                 else if (type == SESMAN_SESSION_TYPE_XVNC)
                 {
-                    env_check_password_file(passwd_file, password);
+                    if (s->guid != 0)
```
Comment (proski) [inline sesman/session.c]: This line causes a warning in clang:
`session.c:700:28: warning: comparison of array 's->guid' not equal to a null pointer is always true [-Wtautological-pointer-compare]`
Which raised a bigger question. Why so many ifs? Shouldn't the new behavior be required?

**Comment by proski** [comment]
I checked the patch on Fedora 25. I cannot reconnect to an existing VNC session. I use FreeRDP without autologin. I get authentication failure.

**Comment by jsorg71** [comment]
@proski I think the VNC reconnect issue is fixed.

**Comment by proski** [comment]
The reconnect issue is fixed. The warning is fixed. But I have found another issue. I cannot login to the Xvnc session if `--enable-ipv6` was used to configure xrdp. I'm connecting from Mac OSX Sierra to RHEL 6. The clients are freerdp, MS RDP and MS RDP Beta. Both manual login and autologin fail.

**Comment by metalefty** [comment]
It works for me.  FreeRDP to CentOS 7.

```
$ rpm -qa |grep xrdp-
xorg-x11-drv-xrdp-0.9.0.git822e3b8+devel-1.el7.centos.x86_64
xrdp-0.9.0.gitf14e3dc+vpf-1.el7.centos.x86_64
```

```
[20161205-16:51:00] [INFO ] A connection received from ::1 port 57936
[20161205-16:51:01] [INFO ] ++ created session (access granted): username osc, ip 0.0.0.0:45798 - socket: 12
[20161205-16:51:01] [INFO ] starting Xvnc session...
[20161205-16:51:01] [DEBUG] Closed socket 9 (AF_INET6 :::5910)
[20161205-16:51:01] [DEBUG] Closed socket 9 (AF_INET6 :::6010)
[20161205-16:51:01] [DEBUG] Closed socket 9 (AF_INET6 :::6210)
[20161205-16:51:01] [DEBUG] Closed socket 7 (AF_INET6 ::1:3350)
[20161205-16:51:01] [DEBUG] Closed socket 8 (AF_INET6 ::1:3350)
[20161205-16:51:01] [INFO ] Xvnc :10 -geometry 1024x768 -depth 16 -rfbauth /home/osc/.vnc/sesman_osc_passwd -bs -nolisten tcp -localhost -dpi 96
[20161205-16:51:01] [INFO ] starting xrdp-sessvc - xpid=2300 - wmpid=2299
```

**Comment by proski** [comment]
I was able to reproduce the VNC authentication problem without this patch, with the unchanged `devel` branch. I'll create a separate issue for that.

**Comment by jsorg71** [comment]
so we are good for this PR? any objections?

**Comment by metalefty** [comment]
LGTM

**Comment by proski** [comment]
No objections on my part. Let's merge.

**Comment by jsorg71** [comment]
I merged this but the github server crashed at the same time but it looks like it merged.

**Pull request merged**